SDWebImage/Core/SDImageCacheDefine.h

```diff
@@ -68,12 +69,14 @@ FOUNDATION_EXPORT UIImage * _Nullable SDImageCacheDecodeImageData(NSData * _Nonn
  @param key The image cache key
  @param options A mask to specify options to use for this query
  @param context A context contains different options to perform specify changes or processes, see `SDWebImageContextOption`. This hold the extra objects which `options` enum can not hold.
+ @param cacheType Specify where to query the cache from. By default we use `.all`, which means both memory cache and disk cache. You can choose to query memory only or disk only as well. Pass `.none` is invalid and callback with nil immediatelly.
  @param completionBlock The completion block. Will not get called if the operation is cancelled
  @return The operation for this query
  */
 - (nullable id<SDWebImageOperation>)queryImageForKey:(nullable NSString *)key
                                              options:(SDWebImageOptions)options
                                              context:(nullable SDWebImageContext *)context
+                                           cacheType:(SDImageCacheType)cacheType
```

Comment:
Does this cause API break ?
If so, maybe in 5.x we need to keep this original method, which translate to the new method, with `caacheType` arg into `.all`

Reply:
In the 6.x we can remove this old API. As well as some other out-dated API

Reply:
@kinarobin I've updated the test cases as well, review again ?

Reply:
Emm.

Reply:
I don't think this will cause API break.

Reply:
Previously (first commit here: fbe76bc43666aaa27b510aeef0ecf2e87e27c3b1) I remove the old API :)